### PR TITLE
net: lib: Fix MDNS RESPONDER

### PR DESCRIPTION
**Pull request by loicpoulain**
1. Fix malformed MDNS response

2. The dns directory was not included when building with
DNS_RESONDER or LLMR_RESPONDER configs.

### CODE REVIEW

File: subsys/net/lib/CMakeLists.txt [hunk 1]
```diff
@@ -5,12 +5,17 @@ add_subdirectory_if_kconfig(coap)
...
 add_subdirectory_if_kconfig(socks)
 add_subdirectory_if_kconfig(sntp)
-add_subdirectory_ifdef(CONFIG_DNS_RESOLVER      dns)
 add_subdirectory_ifdef(CONFIG_MQTT_LIB          mqtt)
 add_subdirectory_ifdef(CONFIG_NET_CONFIG_SETTINGS config)
 add_subdirectory_ifdef(CONFIG_NET_SOCKETS       sockets)
 add_subdirectory_ifdef(CONFIG_TLS_CREDENTIALS   tls_credentials)
 
+if (CONFIG_DNS_RESOLVER
+    OR CONFIG_DNS_RESPONDER
```
Comment (tbursztyka) [inline subsys/net/lib/CMakeLists.txt]: According to the Kconfig, you cannot have CONFIG_DNS_RESPONDER nor CONFIG_LLMNR_RESPONDER without CONFIG_DNS_RESOLVER

What was the config you were trying to apply?

**Review by jukkar — COMMENTED** [review]
According to Kconfig file this should not be needed.
The Kconfig file has
```
config DNS_RESOLVER
	bool "DNS resolver"

if DNS_RESOLVER

config MDNS_RESOLVER
	bool "Enable mDNS support"

config LLMNR_RESOLVER
	bool "Enable LLMNR support"

endif
```

I just wonder should we put `depends on DNS_RESOLVER` to mDNS and LLMNR configs.

**Comment by loicpoulain** [comment]
@tbursztyka , I'm building MDNS sample which has CONFIG_MDNS_RESPONDER (only). 
@jukkar, it's **MDNS_RESPONDER** not RESOLVER, so DNS_RESOLVER should not be a dependency.

**Comment by tbursztyka** [comment]
@loicpoulain So Kconfig should be fixed as well, not only the cmake file.

**Comment by loicpoulain** [comment]
> @loicpoulain So Kconfig should be fixed as well, not only the cmake file.

What do you mean? in Kconfig, MDNS_RESPONDER, is correctly defined out of DNS_RESOLVER if block.

**Comment by loicpoulain** [comment]
ping

**Review by jukkar — APPROVED** [review]

**Pull request merged**